activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -147,6 +151,8 @@ def initialize(connection)
       end
 
       def begin_transaction(options = {})
+        options[:run_callbacks] ||= !current_transaction.joinable?
```

Comment:
yeah.. i was not sure about this API.
I like your suggestion, will do that.. thanks!